x-pack/solutions/observability/plugins/apm/public/components/app/service_map/__stories__/cytoscape_example_data.stories.tsx

```diff
@@ -114,87 +99,84 @@ export const GenerateMap: Story<{}> = () => {
   );
 };
 
-export const MapFromJSON: Story<{}> = () => {
-  const [json, setJson] = useState<string>(
-    getSessionJson() || JSON.stringify(exampleResponseTodo, null, 2)
-  );
-  const [error, setError] = useState<string | undefined>();
+interface MapFromJSONArgs {
+  json: ServiceMapResponse;
+}
+const MapFromJSONTemplate: Story<MapFromJSONArgs> = () => {
+  const [{ json }, updateArgs] = useArgs();
 
+  const [error, setError] = useState<string | undefined>();
   const [elements, setElements] = useState<any[]>([]);
 
   const [uniqueKeyCounter, setUniqueKeyCounter] = useState<number>(0);
-  const updateRenderedElements = () => {
+  const updateRenderedElements = useCallback(() => {
     try {
-      setElements(JSON.parse(json).elements);
+      if ('elements' in json) {
+        setElements(json.elements);
+      } else {
+        const paths = getPaths({ spans: json.spans });
+        const nodes = getServiceMapNodes({
+          anomalies: json.anomalies,
+          connections: paths.connections,
+          servicesData: json.servicesData,
+          exitSpanDestinations: paths.exitSpanDestinations,
+        });
+
+        setElements(nodes.elements);
+      }
       setUniqueKeyCounter((key) => key + 1);
       setError(undefined);
     } catch (e) {
       setError(e.message);
     }
-  };
+  }, [json]);
 
   useEffect(() => {
     updateRenderedElements();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [updateRenderedElements]);
 
   return (
-    <div>
-      <Cytoscape key={uniqueKeyCounter} elements={elements} height={getHeight()}>
-        <Centerer />
-      </Cytoscape>
-      <EuiForm isInvalid={error !== undefined} error={error}>
-        <EuiFlexGroup>
-          <EuiFlexItem>
-            <CodeEditor // TODO Unable to find context that provides theme. Need CODEOWNER Input
-              languageId="json"
-              value={json}
-              options={{ fontFamily: 'monospace' }}
-              onChange={(value) => {
-                setJson(value);
-                setSessionJson(value);
-              }}
-            />
-          </EuiFlexItem>
```

Comment:
Storybook args does the same thing.

Reply:
Nit: Maybe it's a good idea to mention that instead of uploading the file, we can also use the controls, as removing this field can be confusing (looks like removing the option to paste the JSON directly) -> just a short paragraph saying that this can be done now by changing it in the controls would be enough, wdyt?